### PR TITLE
Gradle 7, Java 16 build compatibility and Github Actions

### PR DESCRIPTION
**Pull request by altrisi**
This PR updates Gradle to 7 and Loom to 0.7, allowing to build the mod with Java 16, and adds Github Actions to build and create those green checks on push and in pull requests, so you make sure you are not merging broken code.

Also cleans up a few of the commented lines in build.gradle, can revert if you want those.

### CODE REVIEW

**Comment by altrisi** [comment]
Actions also add a nice ci where people can download development builds if they want.

**Comment by replaceitem** [comment]
Thanks for cleaning up my gradle mess ;)
Github actions also appreciated

File: build.gradle [hunk 1]
```diff
@@ -46,7 +46,7 @@ dependencies {
 	//modCompile "io.vavr:vavr:0.10.1"
 	include "io.vavr:vavr:0.10.1"
 
-	modCompile "com.vdurmont:emoji-java:5.1.1"
```
Comment (replaceitem) [inline build.gradle:49]: Whats the difference between those?
just curious

**Comment by altrisi** [comment]
It's just that Gradle 6 or 7 removed them and `implementation` is basically the equivalent in newer Gradle versions.

**Comment by replaceitem** [comment]
Ah ok makes sense

**Pull request merged**